pkg/controllermanager/controller/namespacedcloudprofile/reconciler.go

```diff
@@ -140,11 +141,29 @@ func mergeExpirationDates(base, override gardencorev1beta1.ExpirableVersion) gar
 }
 
 func mergeMachineImages(base, override gardencorev1beta1.MachineImage) gardencorev1beta1.MachineImage {
-	base.Versions = mergeDeep(base.Versions, override.Versions, machineImageVersionKeyFunc, mergeMachineImageVersions, true)
-	return base
+	if override.UpdateStrategy == nil {
+		// The NamespacedCloudProfile only extends parent CloudProfile machine image versions by the expiration date.
+		base.Versions = mergeDeep(base.Versions, override.Versions, machineImageVersionKeyFunc, mergeMachineImageVersions, true)
+		return base
+	}
+
+	// A formerly new machine image in the NamespacedCloudProfile has been added to the parent CloudProfile by now.
+	// If NamespacedCloudProfile and parent CloudProfile UpdateStrategies conflict, only use NamespacedCloudProfile machine images.
+	if override.UpdateStrategy == base.UpdateStrategy {
```

Comment:
Why is this only done when the update strategy is equal? Most of the times they should be, unless someone configured their NamespacedCloudProfile wrongly.

Reply:
I rely on this single assertion because the `UpdateStrategy` is the only attribute of `MachineImage` besides the `Name` (must be equal anyway) and the `Versions` slice.

So if the `UpdateStrategy` is (a) given and (b) equal, I can go ahead and also merge the existing machine image versions from the parent cloud profile to the image versions in the `NamespacedCloudProfile`.

Someone should not be able to configure their `NamespacedCloudProfile` wrongly, as for an extended (i.e. already existing in the parent cloud profile) machine image no custom `UpdateStrategy` may be provided.